content/en/apps/examples/covid-rdt-reference-app.md

```diff
@@ -0,0 +1,490 @@
+ * **[Postgres queries](https://github.com/medic/cht-couch2pg):** CHT ships with a utility to export all the data that the API has to a relational database, Postgres. You have all the raw data the API has, but can now use the power of joins and groupings to come up with totally customizable stats by day, week, month etc. Data can be synched daily or hourly from the CHT. 
+ * **3rd party integrations:** We have used both [Klipfolio](https://www.klipfolio.com/) and, more recently, [Superset](https://superset.incubator.apache.org/) to create more user-friendly dashboards than any of the above options offer.  They talk to the Postgres server as the back end.
+
+
+### Sample API call
+
+All calls to the API need to be authenticated. Be sure to follow our documentation on how to do this, but otherwise you need to use a login with `admin` permissions with this stucture: `http://LOGIN:PASSWORD@HOSTNAME`. If you had deployed this application without any customizations, you would have these forms available as seen by calling the [forms API]({{< ref "apps/reference/api#get-apiv1forms" >}}) (`/api/v1/forms`):
+
+* contact:clinic:create.xml
+* contact:clinic:edit.xml
+* contact:district_hospital:create.xml
+* contact:district_hospital:edit.xml
+* contact:health_center:create.xml
+* contact:health_center:edit.xml
+* contact:person:create.xml
+* contact:person:edit.xml
+* covid19_rdt_capture.xml
+* covid19_rdt_provision.xml
+
+The last two forms can be used to query [the reports API ]({{< ref "apps/reference/api#get-apiv2exportreports" >}}) to get RDT provions and captures.  To get all the capture reports into a file called `output.csv`, you would use this call:
+
+```shell
+curl "http://LOGIN:PASSWORD@HOSTNAME/api/v2/export/reports?filters[search]=&filters[forms][selected][0][code]=covid19_rdt_capture" > output.csv
+```
+
+### Base64 image extraction
+
+To extract the binary image from the ASCII value in a report, you can view it in the CHT or you use a script to do so.  Here's an example of using Bash utilities (`sed`, `cut`, `tr` and `base64`) in Linux to get the image in the 1st row of data after the column headers into the file `image.jpg`. 
+
+```shell
+sed '2q;d' output.csv | cut -d',' -f15 | tr -d '"' | base64 -d > image.jpg
+```
+
+When retrieving [JSON](#capture-1), this value is found in `capture.android-app-outputs.rdt_session_bundle.rdt_session_result_bundle.rdt_session_result_extra_images.cropped` field.
+
+{{% alert title="Note" %}}The reports API always outputs in CSV, but the COVID-19 application uses [Base64 encoding](https://en.wikipedia.org/wiki/Base64) to store the images as text as noted above.  These may misbehave when opened them in a spreadsheet application like LibreOffice ("maximum number of characters per cell exceeded") or Excel (silently clipped to 32k chars) as they're thousands, if not hundreds of thousands, of characters long.  Be sure to programmatically process these into image files as needed. {{% /alert %}}
```

Comment:
Do you have a script that you used internally already for this? It could be useful to post a command or snippet as an example rather than leaving it as an exercise from scratch.

Reply:
This is actually what I did!  The call to the reports API above results in a `outputs.csv` file.  The shell call in this section uses that same file to generate an image from one of the rows.

I don't think it's useful to bulk export the images from a CSV file (say in a while loop in Bash) as likely you'd be doing either an ETL into a 3rd party system, or just raw bulk import w/ no ETL and then on display show the base64 as an image, like we do with the CHT.

Maybe this wasn't clear though and I could add some extra explanation about this? Maybe you disagree and that a `while` script in Bash here is what's needed?

Reply:
Right, not entirely _from scratch_, but from the command to extract a single image. It is likely not obvious from the extraction command what to change to get the other images out for a test, so to make it easier I think including the while loop would be helpful. If you think that there are too many aspects to take into account, then another option would be to mention which part to change to extract images from different lines and then if someone needs to do it programmatically they can figure it out from there.

Also, your comment about ETL is useful, and it could be worth mentioning in the text to give proper context.

Reply:
Oh - cool cool! thanks for the push-back on the push-back! I've tended to this in b71879f and b24dc48.  Lemme know what ya think!

Reply:
Nice! Looks great!!